File: app/src/main/java/com/routee/qianbaotest/model/bean/UserInfo.java
package com.routee.qianbaotest.model.bean;

/**
 * Created by hdb on 2017/8/7.
 * description: ${TODO}
 */

public class UserInfo {
    private String userName;
    private static UserInfo sUserInfo = null;

    private UserInfo() {
    }

    public static UserInfo getInstance() {
        synchronized (UserInfo.class) {
            if (sUserInfo == null) {
                sUserInfo = new UserInfo();
            }
        }
        return sUserInfo;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }
}

File: app/src/main/java/com/routee/qianbaotest/model/bean/UserInfo.kt
package com.routee.qianbaotest.model.bean

/**
 * Created by hdb on 2017/8/7.
 * description: ${TODO}
 */

class UserInfo private constructor() {
    var userName: String? = null

    companion object {
        private var sUserInfo: UserInfo? = null

        val instance: UserInfo
            get() {
                synchronized(UserInfo::class.java) {
                    if (sUserInfo == null) {
                        sUserInfo = UserInfo()
                    }
                }
                return this!!.sUserInfo!!
            }
    }
}
